junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/ClassTestDescriptor.java

```diff
@@ -201,11 +205,70 @@ private Object instantiateAndPostProcessTestInstance(JupiterEngineExecutionConte
 		return instance;
 	}
 
+	protected Object invokeTestInstanceConstructor(Object outerInstance, ExtensionRegistry registry,
+			ExtensionContext extensionContext) {
+		Constructor<?> constructor = ReflectionUtils.getDeclaredConstructor(this.testClass);
+		if (outerInstance != null) {
+			return executableInvoker.invoke(constructor, outerInstance, extensionContext, registry);
+		}
+		else {
+			return executableInvoker.invoke(constructor, extensionContext, registry);
+		}
+	}
+
+	protected TestInstanceFactory resolveTestInstanceFactory(ExtensionRegistry registry,
+			ExtensionRegistry parentRegistry) {
+
+		List<TestInstanceFactory> factories = registry.getExtensions(TestInstanceFactory.class);
+		if (factories.isEmpty()) {
+			return null;
+		}
+
+		if (parentRegistry != null) {
+			List<TestInstanceFactory> parentFactories = parentRegistry.getExtensions(TestInstanceFactory.class);
+			factories.removeAll(parentFactories);
```

Comment:
If we are going to ignore factories registered in parent contexts -- if and only if there is a locally registered factory --  we then need to explicitly document that behavior.

Reply:
I have clarified the behavior in both the javadoc and user guide.

Reply:
ok. thanks